gui/gm-unit.lua

```diff
@@ -612,7 +612,129 @@ function editor_wounds:init( args )
+        end
+    end
+    for i,v in ipairs(m_attrs) do
+        local attr_name=format_attr(df.mental_attribute_type[i],max_len)
+        local text=string.format("%s: %d/%d",
+            attr_name,v.value,v.max_value)
+        table.insert(ret,{
+            text=text,
+            attr=v,
+            attr_name=attr_name,
+            search_key=text:lower()
+        })
+    end
+    for i,v in ipairs(b_attrs) do
+        local attr_name=format_attr(df.physical_attribute_type[i],max_len)
+        local text=string.format("%s: %d/%d",
+            attr_name,v.value,v.max_value)
+        table.insert(ret,{
+            text=text,
+            attr=v,
+            attr_name=attr_name,
+            search_key=text:lower()
+        })
+    end
+    return ret
+end
+function editor_attrs:update_list(no_save_place)
+    local attr_list=list_attrs(self.target_unit)
+    if no_save_place then
+        self.subviews.attributes:setChoices(attr_list)
+    else
+        self.subviews.attributes:setChoices(attr_list,self.subviews.attributes:getSelected())
+    end
+end
+
+function editor_attrs:init( args )
+    if self.target_unit.status.current_soul==nil then
+        qerror("Unit does not have soul, can't edit mental attributes")
+    end
+
+    local attr_list=list_attrs(self.target_unit)
 
+    self:addviews{
+        widgets.FilteredList{
+            choices=attr_list,
+            frame = {t=0, b=1,l=1},
+            view_id="attributes",
+        },
+        widgets.Label{
+            frame = { b=0,l=1},
+            text ={{text= ": exit editor ",
+                key  = "LEAVESCREEN",
+                on_activate= self:callback("dismiss")
+                },
+                {text=": set max attribute ",
+                key = "SEC_SELECT",
+                on_activate= function (  )
+                    local a,a_name=self:get_cur_attr()
+                    dialog.showInputPrompt(a_name,"Enter new max value:",COLOR_WHITE,
+                    tostring(a.max_value),function(new_value)
+                        a.max_value=new_value
+                        self:update_list()
+                    end)
+                end
+                },
+                {text=": set attribute ",
+                key = "SELECT",
+                on_activate= function (  )
+                    local a,a_name=self:get_cur_attr()
+                    dialog.showInputPrompt(a_name,"Enter new value:",COLOR_WHITE,
+                    tostring(a.value),function(new_value)
+                        a.value=new_value
+                        self:update_list()
+                    end)
+                end
+                }
+            }
+        },
+    }
+end
+function editor_attrs:get_cur_attr()
+    local list_wid=self.subviews.attributes
+    local _,choice=list_wid:getSelected()
+    if choice==nil then
+        qerror("Nothing selected")
+    end
+    return choice.attr,choice.attr_name
+end
+function editor_attrs:remove_rust(attr)
+    --TODO
```

Comment:
Is this something you're planning on addressing or is this good to be merged?

Reply:
It can be merged, if someone wants (or asks) then derusting can be added.